src/jit/morph.cpp

```diff
@@ -10855,10 +10850,7 @@ GenTreePtr          Compiler::fgMorphSmpOp(GenTreePtr tree, MorphAddrContext* ma
                 }
                 else if (andOp->gtType == TYP_LONG)
                 {
-                    if (andOp->gtOp.gtOp2->gtOper != GT_CNS_NATIVELONG)
-                        goto SKIP;
-
-                    if (andOp->gtOp.gtOp2->gtIntConCommon.LngValue() != 1)
+                    if (!andOp->IsIOne())
```

Comment:
GT_CNS_NATIVELONG is defined as GT_CNS_INT on 64-bit and GT_NS_LNG on32-bit targets.  Once again the above check is more liberal than required and not giving the right picture to a person who reads this code.